### ISSUE TITLE
Retrieving BelongsTo model overwriting relationship foreign key

### ISSUE THREAD

**Issue by JonForest** [digitalsadhu/loopback-component-jsonapi#97]
When I retrieve models with a belongsTo attribute, the linked foreign key Id is being overwritten in the database.  I'm only making GET requests.

My best description of the issue is below, sorry I don't have the familiarity to really explain this well.
I have two models: item and listItem.
'item' has one property - name
'listItem' has one property - qty, and a belongsTo relationship with item
So, the listitem table looks like
```
id  qty  itemId
10  1    1
11  1    2
```
When I retrieve all items and listItems it makes the following GET requests
GET http://localhost:3000/api/listitems/10/item
GET http://localhost:3000/api/listitems/11/item
GET http://localhost:3000/api/items
http://localhost:3000/api/listitems

The table now looks like
```
id  qty  itemId
10  1    2
11  1    2
```
For the sake of example, if I run 'update listitem set itemId=0;'
And then replay on the /listitems/ XHR requests, it sets that itemId to 2.
E.g.
```
id  qty  itemId
10  1    0
11  1    0
```

If I then replay the following XHR
GET http://localhost:3000/api/listitems/10/item
The table updates to
```
id  qty  itemId
10  1    2
11  1    0
```
I can see the UPDATE happening in the SQL logs, but am not able to track down in the loopback code the logic that is making it fire.

**Comment by digitalsadhu** [comment]
Thanks for reporting @JonForest 
We should investigate further. 

Have you tried playing around with loopback debug strings?
https://docs.strongloop.com/display/public/LB/Setting+debug+strings
Perhaps you see if you can figure out more about where the save is happening using those?

Can you also verify in the chrome network tab that theres no way the ember client is sending any posts or patches?

Do you use any hooks on your models? Model.observes('access') etc. Is there any way you might be doing a save or update that could be being triggered?

I've also just gone back and done a bit of a code audit for where this might be happening and I have a likely candidate. Could you throw a couple breakpoints [here](https://github.com/digitalsadhu/loopback-component-jsonapi/blob/master/lib/relationships.js#L76) and [here](https://github.com/digitalsadhu/loopback-component-jsonapi/blob/master/lib/relationships.js#L127) and see if they get hit when you run your GET requests?

**Comment by digitalsadhu** [comment]
Also, is there any chance you are sending any data with your GET requests from the client?

**Comment by JonForest** [comment]
Definitely not sending anything other than GET requests, and can in fact recreate from a command line curl requests.
e.g.
```
curl 'http://localhost:3000/api/listitems/2/item'
```
It's hitting [L127](https://github.com/digitalsadhu/loopback-component-jsonapi/blob/master/lib/relationships.js#L127) on this request.

I'll spend some more time as I'm able today to dig into this a little further and see if I can get to the bottom of it.

**Comment by digitalsadhu** [comment]
Ok great thanks. The fact that it's hitting line 27 means I know how to fix. Will do so when I get a chance

**Comment by digitalsadhu** [comment]
Actually is there any chance you can have a go at submitting a failing test @JonForest?

**Comment by JonForest** [comment]
Yep, I'll give it a bash a bit later.

**Comment by JonForest** [comment]
A rather ugly set of tests, but it gives you the idea.  
https://github.com/digitalsadhu/loopback-component-jsonapi/pull/101
Interesting that it works when the models are set up via code, but fails when set up via XHRs.

**Comment by digitalsadhu** [comment]
Thanks man, I'm pretty sure I know the fix, but really great to have the test to prove it. Will see if I can get to that soon.

**Comment by digitalsadhu** [comment]
Fixed in v0.16.2
Thanks man!
